### ISSUE TITLE
/bin/ld: /usr/local/lib/libcblas.a(cblas_sgemm.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC

### ISSUE THREAD

**Issue by lengyuyeke** [dmlc/mxnet#5454]
I install mxnet on linux ,when I ran make -j$(nproc)
,I got the error below ,and I can't find any solution on the internet 
`
g++ -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -I/data/lipengfei/software/mxnet/mshadow/ -I/data/lipengfei/software/mxnet/dmlc-core/include -fPIC -I/data/lipengfei/software/mxnet/nnvm/include -Iinclude -funroll-loops -Wno-unused-variable -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_CUDA=0 -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/local/include/opencv -I/usr/local/include   -fopenmp  -DMXNET_USE_NVRTC=0 -shared -o lib/libmxnet.so build/src/operator/mkl/mkl_cppwrapper.o build/src/operator/mkl/mkl_memory.o build/src/operator/tensor/elemwise_binary_broadcast_op_extended.o build/src/operator/tensor/elemwise_binary_op_extended.o build/src/operator/tensor/init_op.o build/src/operator/tensor/broadcast_reduce_op_index.o build/src/operator/tensor/elemwise_binary_scalar_op_logic.o build/src/operator/tensor/elemwise_binary_broadcast_op_basic.o build/src/operator/tensor/broadcast_ra - build/src/c_api/c_api_executor.o
educe_op_value.o build/src/operator/tensor/control_flow_op.o build/src/operator/tensor/elemwise_binary_op_basic.o build/src/operator/tensor/elemwise_unary_op.o build/src/operator/tensor/elemwise_sum.o build/src/operator/tensor/elemwise_binary_scalar_op_extended.o build/src/operator/tensor/indexing_op.o build/src/operator/tensor/ordering_op.o build/src/operator/tensor/elemwise_binary_broadcast_op_logic.o build/src/operator/tensor/sample_op.o build/src/operator/tensor/elemwise_binary_op_logic.o build/src/operator/tensor/elemwise_binary_scalar_op_basic.o build/src/operator/tensor/matrix_op.o build/src/operator/nnpack/nnpack_util.o build/src/io/io.o build/src/io/image_aug_default.o build/src/io/iter_csv.o build/src/io/image_io.o build/src/io/iter_image_recordio.o build/src/io/iter_mnist.o build/src/common/mxrtc.o build/src/nnvm/legacy_op_util.o build/src/nnvm/legacy_json_util.o build/src/ndarray/ndarray_function.o build/src/ndarray/ndarray.o build/src/operator/instance_norm.o build/src/operator/svm_output.o build/src/operator/native_op.o build/src/operator/pooling.o build/src/operator/crop.o build/src/operator/spatial_transformer.o build/src/operator/deconvolution.o build/src/operator/swapaxis.o build/src/operator/batch_norm.o build/src/operator/softmax_output.o build/src/operator/operator_util.o build/src/operator/sequence_reverse.o build/src/operator/pad.o build/src/operator/sequence_last.o build/src/operator/make_loss.o build/src/operator/operator.o build/src/operator/optimizer_op.o build/src/operator/lrn.o build/src/operator/correlation.o build/src/operator/fully_connected.o build/src/operator/sequence_mask.o build/src/operator/grid_generator.o build/src/operator/convolution.o build/src/operator/identity_attach_KL_sparse_reg.o build/src/operator/bilinear_sampler.o build/src/operator/dropout.o build/src/operator/cudnn_batch_norm.o build/src/operator/loss_binary_op.o build/src/operator/activation.o build/src/operator/regression_output.o build/src/operator/l2_normalization.o build/src/operator/upsampling.o build/src/operator/concat.o build/src/operator/leaky_relu.o build/src/operator/roi_pooling.o build/src/operator/ndarray_op.o build/src/operator/custom.o build/src/operator/cudnn_convolution.o build/src/operator/slice_channel.o build/src/operator/cross_device_copy.o build/src/operator/softmax_activation.o build/src/operator/rnn.o build/src/engine/profiler.o build/src/engine/naive_engine.o build/src/engine/threaded_engine_pooled.o build/src/engine/threaded_engine.o build/src/engine/engine.o build/src/engine/threaded_engine_perdevice.o build/src/storage/storage.o build/src/c_api/c_api_executor.o build/src/c_api/c_api_symbolic.o build/src/c_api/c_api_ndarray.o build/src/c_api/c_predict_api.o build/src/c_api/c_api.o build/src/c_api/c_api_error.o build/src/executor/inplace_addto_detect_pass.o build/src/executor/graph_executor.o build/src/executor/attach_op_execs_pass.o build/src/executor/attach_op_resource_pass.o build/src/kvstore/kvstore.o build/src/resource.o build/src/initialize.o /data/lipengfei/software/mxnet/dmlc-corea - build/src/c_api/c_api_symbolic.o
/libdmlc.a -pthread -lm -lcblas -fopenmp -lrt -L/usr/local/lib -lopencv_ml -lopencv_objdetect -lopencv_shape -lopencv_stitching -lopencv_superres -lopencv_videostab -lopencv_calib3d -lopencv_features2d -lopencv_highgui -lopencv_videoio -lopencv_imgcodecs -lopencv_video -lopencv_photo -lopencv_imgproc -lopencv_flann -lopencv_core   -L/usr/include/atlas \
 -Wl,--whole-archive /data/lipengfei/software/mxnet/nnvm/lib/libnnvm.a -Wl,--no-whole-archive
a - build/src/c_api/c_api_ndarray.o
a - build/src/c_api/c_predict_api.o
a - build/src/c_api/c_api.o
a - build/src/c_api/c_api_error.o
a - build/src/executor/inplace_addto_detect_pass.o
a - build/src/executor/graph_executor.o
a - build/src/executor/attach_op_execs_pass.o
a - build/src/executor/attach_op_resource_pass.o
a - build/src/kvstore/kvstore.o
a - build/src/resource.o
a - build/src/initialize.o
g++ -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -I/data/lipengfei/software/mxnet/mshadow/ -I/data/lipengfei/software/mxnet/dmlc-core/include -fPIC -I/data/lipengfei/software/mxnet/nnvm/include -Iinclude -funroll-loops -Wno-unused-variable -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -DMSHADOW_USE_CUDA=0 -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/local/include/opencv -I/usr/local/include   -fopenmp  -DMXNET_USE_NVRTC=0 -std=c++11  -o bin/im2rec tools/im2rec.cc build/src/operator/mkl/mkl_cppwrapper.o build/src/operator/mkl/mkl_memory.o build/src/operator/tensor/elemwise_binary_broadcast_op_extended.o build/src/operator/tensor/elemwise_binary_op_extended.o build/src/operator/tensor/init_op.o build/src/operator/tensor/broadcast_reduce_op_index.o build/src/operator/tensor/elemwise_binary_scalar_op_logic.o build/src/operator/tensor/elemwise_binary_broadcast_op_basic.o build/src/operator/tensor/broadcast_reduce_op_value.o build/src/operator/tensor/control_flow_op.o build/src/operator/tensor/elemwise_binary_op_basic.o build/src/operator/tensor/elemwise_unary_op.o build/src/operator/tensor/elemwise_sum.o build/src/operator/tensor/elemwise_binary_scalar_op_extended.o build/src/operator/tensor/indexing_op.o build/src/operator/tensor/ordering_op.o build/src/operator/tensor/elemwise_binary_broadcast_op_logic.o build/src/operator/tensor/sample_op.o build/src/operator/tensor/elemwise_binary_op_logic.o build/src/operator/tensor/elemwise_binary_scalar_op_basic.o build/src/operator/tensor/matrix_op.o build/src/operator/nnpack/nnpack_util.o build/src/io/io.o build/src/io/image_aug_default.o build/src/io/iter_csv.o build/src/io/image_io.o build/src/io/iter_image_recordio.o build/src/io/iter_mnist.o build/src/common/mxrtc.o build/src/nnvm/legacy_op_util.o build/src/nnvm/legacy_json_util.o build/src/ndarray/ndarray_function.o build/src/ndarray/ndarray.o build/src/operator/instance_norm.o build/src/operator/svm_output.o build/src/operator/native_op.o build/src/operator/pooling.o build/src/operator/crop.o build/src/operator/spatial_transformer.o build/src/operator/deconvolution.o build/src/operator/swapaxis.o build/src/operator/batch_norm.o build/src/operator/softmax_output.o build/src/operator/operator_util.o build/src/operator/sequence_reverse.o build/src/operator/pad.o build/src/operator/sequence_last.o build/src/operator/make_loss.o build/src/operator/operator.o build/src/operator/optimizer_op.o build/src/operator/lrn.o build/src/operator/correlation.o build/src/operator/fully_connected.o build/src/operator/sequence_mask.o build/src/operator/grid_generator.o build/src/operator/convolution.o build/src/operator/identity_attach_KL_sparse_reg.o build/src/operator/bilinear_sampler.o build/src/operator/dropout.o build/src/operator/cudnn_batch_norm.o build/src/operator/loss_binary_op.o build/src/operator/activation.o build/src/operator/regression_output.o build/src/operator/l2_normalization.o build/src/operator/upsampling.o build/src/operator/concat.o build/src/operator/leaky_relu.o build/src/operator/roi_pooling.o build/src/operator/ndarray_op.o build/src/operator/custom.o build/src/operator/cudnn_convolution.o build/src/operator/slice_channel.o build/src/operator/cross_device_copy.o build/src/operator/softmax_activation.o build/src/operator/rnn.o build/src/engine/profiler.o build/src/engine/naive_engine.o build/src/engine/threaded_engine_pooled.o build/src/engine/threaded_engine.o build/src/engine/engine.o build/src/engine/threaded_engine_perdevice.o build/src/storage/storage.o build/src/c_api/c_api_executor.o build/src/c_api/c_api_symbolic.o build/src/c_api/c_api_ndarray.o build/src/c_api/c_predict_api.o build/src/c_api/c_api.o build/src/c_api/c_api_error.o build/src/executor/inplace_addto_detect_pass.o build/src/executor/graph_executor.o build/src/executor/attach_op_execs_pass.o build/src/executor/attach_op_resource_pass.o build/src/kvstore/kvstore.o build/src/resource.o build/src/initialize.o /data/lipengfei/software/mxnet/dmlc-core/libdmlc.a /data/lipengfei/software/mxnet/nnvm/lib/libnnvm.a -pthread -lm -lcblas -fopenmp -lrt -L/usr/local/lib -lopencv_ml -lopencv_objdetect -lopencv_shape -lopencv_stitching -lopencv_superres -lopencv_videostab -lopencv_calib3d -lopencv_features2d -lopencv_highgui -lopencv_videoio -lopencv_imgcodecs -lopencv_video -lopencv_photo -lopencv_imgproc -lopencv_flann -lopencv_core   -L/usr/include/atlas
/bin/ld: /usr/local/lib/libcblas.a(cblas_sgemm.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/local/lib/libcblas.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
make: *** [lib/libmxnet.so] Error 1
make: *** Waiting for unfinished jobs....
/usr/local/lib/libcblas.a(sdotsub.o): In function `sdotsub_':
sdotsub.f:(.text+0x7): undefined reference to `sdot_'
/usr/local/lib/libcblas.a(ddotsub.o): In function `ddotsub_':
ddotsub.f:(.text+0x7): undefined reference to `ddot_'
/usr/local/lib/libcblas.a(cblas_sgemm.o): In function `cblas_sgemm':
cblas_sgemm.c:(.text+0x138): undefined reference to `sgemm_'
cblas_sgemm.c:(.text+0x1fe): undefined reference to `sgemm_'
/usr/local/lib/libcblas.a(cblas_dgemm.o): In function `cblas_dgemm':
cblas_dgemm.c:(.text+0x140): undefined reference to `dgemm_'
cblas_dgemm.c:(.text+0x20a): undefined reference to `dgemm_'
collect2: error: ld returned 1 exit status
make: *** [bin/im2rec] Error 1
`
